docs/source/howto/codes.rst

```diff
@@ -4,41 +4,264 @@
+
+Next in lines 12-15 we are creating and setting up the ``codeinfo = CodeInfo()`` object.
+The ``code_uuid`` being passed in line 13 is necessary for the engine to get the base information from the |Code| node (in which computer the code is placed, what is the location of the executable, etc).
+Note that this was taken in the unloading block from ``input_code = self.inputs['code']``, which we never specified in the ``define`` method: this is one of the inputs defined in the base |CalcJob| class that we mentioned earlier when discussing the ``super().define(spec)``:
+
+.. code-block:: python
+
+    spec.input('code', valid_type=orm.Code, help='The `Code` to use for this job.')
+
+Moreover, as this information is necessary for the engine, this input has a ``required=True`` setting (the default for all inputs, including the ones we manually defined earlier).
+For the plugin user this means they will have to provide it when instantiating the calculation (as can be seen in the :ref:`respective section<how-to:codes:run>` below), whereas for the plugin developer this means they will have to make sure to manually pass its UUID from the ``self.inputs['code']`` node to the ``codeinfo.code_uuid`` property, as shown here.
+
+The other two lines are configuring how to build the running script: line 14 indicates where to redirect the standard output, whereas line 15 lists the command line arguments to be passed to the code.
+The specific combination presented here, together with the information inside of the ``code`` node, results in the following script:
+
+.. code-block:: bash
+
+    #!/bin/bash
+
+    '[executable path in code node]' '[input_filename]' > '[output_filename]'
+
+Through the |CodeInfo| object you can also pass more flags (by adding them as string elements to the list in ``codeinfo.cmdline_params``), configure what to pass through the standard input (just as it is shown for the standard output), add commands to be run before and after the execution line, etc.
+
+Finally, the last block remaining in lines 17-19 creates the |CalcInfo| object, passes to it the |CodeInfo| object, and adds the output to the ``retrieve_list``.
+This is a list of all files that the engine needs to copy back from the computer where the code is located, either for immediate parsing or local storage in an output node labeled ``retrieved`` for future post-processing (again, this is another base ``output`` defined in the base |CalcJob| class).
+You can expand this list to include more files, and there are other lists available that allow you to easily specify which files to make available in the remote folder without having to manually copy them during this step and thus having to store unnecessary copies of them in the calculation node.
```

Comment:
This is a *very long* sentence and I find it quite confusing. You are mentioning this and other lists (without actually specifying them) to let the user know how to control what is copied and what is not copied? I'd appreciate if you could try to reword this sentence. It may be warranted its own paragraph.

Reply:
Ok,, removed the clarification for expansion (it is probably self evident) and shortened a bit the phrase about the other lists:


This is a list of all files that the engine needs to copy back from the computer where the code is located, either for immediate parsing or local storage in an output node labeled ``retrieved`` for future post-processing.
The ``retrieved`` node is a ``output`` defined in the base |CalcJob| class.
There are other lists available that allow you to easily customize how to move files to and from the remote working directory in order to prevent the creation of unnecessary copies.